move "dbt models and Dagster software-defined assets" from tutorial t…

sryza opened the pull request:
…o reference

## Summary & Motivation

The beginning of the dbt tutorial currently includes a section titled "dbt models and Dagster software-defined assets", which explains the parallels between dbt models and SDAs. This is great content, but I think it's a lot to digest at once for someone who's opted for the tutorial because they want a gentle walkthrough. It exposes them to these in a short span: the Python API for defining an asset, the term "Software-defined asset", the term "operation", and the term "asset dependency".

This PR proposes moving this content to the reference section of the dbt docs.

## How I Tested These Changes

erinkcochran87 reviewed (commented):
Generally I'm good with this, but I have a few thoughts:

1. Should we link to the section in the reference for this, in case people are curious about this at the start of the tutorial?
2. Should we add a sentence about SDAs in here to at least touch on the concept? We don't need to explain it in-depth, but it could help clarify what the tutorial will end up doing.

Thoughts?

sryza commented:
@erinkcochran87 I incorporated your suggestions and also added an image of the final asset graph we'll end up with. I also changed a sentence inside the dbt model / SDA explanation.

erinkcochran87 reviewed (approved):
Thanks Sandy! :shipit:
